mixer/tools/codegen/pkg/interfacegen/testdata/ReportTmpl.proto

```diff
@@ -39,4 +39,48 @@ message Template {
     google.protobuf.Timestamp timeStamp = 9;
 
     google.protobuf.Duration duration = 10;
+
+    repeated Resource3 res3_list = 11;
+    map<string, Resource3> res3_map = 12;
+}
+
+message Resource1 {
```

Comment:
should `Resource1` be referenced anywhere?  what is this testing?  Is this just verifying that all messages within the proto file get types and instance params, etc. ?

Reply:
So, every message in the proto will have an equivalent *InstanceParam and *Type. For simplicity I choose to generate these artifacts for every message irrespective if they are reachable from the parent 'Template' message or not. We can certainly traverse from the Template and only include what is reachable, however, doing it the existing way is simple and I  hope people just don't write dead messages (not referenced from anywhere). The Template developer should be a reasonable person :-), and will fix the dead message if he/she accidentally defines any.

Thoughts ?

Reply:
As you stated in your comment, this PR and the test is testing what gets generated for Go Instances, Type and InstanceParam messages.